docs/api/event/event.go

```diff
@@ -0,0 +1,158 @@
+package event
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	pollevent "sigs.k8s.io/cli-utils/pkg/kstatus/polling/event"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// Event describes something that has happened during the process or either
+// applying resources or removing resources from the cluster.
+// The Type property defines which event has taken place and the
+// corresponding property will contain more information about the details.
+type Event struct {
+	// Type is the type of event.
+	Type Type
+
+	// InitEvent contains information about which resources will
+	// be applied/pruned.
+	InitEvent InitEvent
+
+	// ErrorEvent contains information about any errors encountered.
+	ErrorEvent ErrorEvent
+
+	// ApplyEvent contains information about progress pertaining to
+	// applying a resource to the cluster.
+	ApplyEvent ApplyEvent
+
+	// StatusEvents contains information about the status of one of
+	// the applied resources.
+	StatusEvent StatusEvent
+
+	// PruneEvent contains information about objects that have been
+	// pruned.
+	PruneEvent PruneEvent
+
+	// DeleteEvent contains information about object that have been
+	// deleted.
+	DeleteEvent DeleteEvent
+}
+
+// Type defines the go type used to enumerate the possible types of events.
+type Type int
+
+const (
+	InitType Type = iota
+	ErrorType
+	ApplyType
+	StatusType
+	PruneType
+	DeleteType
+)
+
+// InitEvent describes the resources that will be applied, pruned or deleted
+// as part of either an apply or destroy operation.
+type InitEvent struct {
+	ActionGroups []ActionGroup
+}
+
+// ResourceAction defines the go type used to enumerate the different actions
+// that can be described in the Init event.
+type ResourceAction int
+
+const (
+	ApplyAction ResourceAction = iota
+	PruneAction
+	DeleteAction
+)
```

Comment:
Would it make sense to add a `StatusAction` here? Or if not is there another plan for how to expose the status functionality?